src/System.Management.Automation/help/CommandHelpProvider.cs

```diff
@@ -511,16 +512,20 @@ private string GetHelpFile(string helpFile, CmdletInfo cmdletInfo)
                     // we have to search only in the application base for a mshsnapin...
                     // if you create an absolute path for helpfile, then MUIFileSearcher
                     // will look only in that path.
-                    helpFileToLoad = Path.Combine(mshSnapInInfo.ApplicationBase, helpFile);
+                    
+                    searchPaths.Add(mshSnapInInfo.ApplicationBase);
+                    searchPaths.Add(HelpUtils.GetUserHomeHelpSearchPath());
                 }
                 else if (cmdletInfo.Module != null && !string.IsNullOrEmpty(cmdletInfo.Module.Path))
-                {
-                    helpFileToLoad = Path.Combine(cmdletInfo.Module.ModuleBase, helpFile);
+                {                    
+                    searchPaths.Add(cmdletInfo.Module.ModuleBase);
+                    searchPaths.Add(HelpUtils.GetModuleBaseForUserHelp(cmdletInfo.Module.ModuleBase, cmdletInfo.Module.Name));
                 }
                 else
                 {
                     searchPaths.Add(GetDefaultShellSearchPath());
                     searchPaths.Add(GetCmdletAssemblyPath(cmdletInfo));
+                    searchPaths.Add(HelpUtils.GetUserHomeHelpSearchPath());
```

Comment:
Does the ordering matter here?  If it finds help in the DefaultShellSearchPath, does it stop searching further?  If so, I think the user home help path should be first as that's most likely to be up-to-date

Reply:
Updated to give preference to user scope path.